app/controllers/application_controller.rb

```diff
@@ -72,9 +69,11 @@ def pundit_unverified_classes
   end
 
   def build_search_cache
-    search_params = {}
-    form_search_params = params.for(ArticleSearchForm)[:article_search_form]
-    search_params.merge!(form_search_params) if form_search_params.is_a?(Hash)
+    if params[:article_search_form]
+      search_params = params[:article_search_form].permit(ArticleSearchForm.new.attributes.keys)
```

Comment:
It would not be necessary to explicitly `.permit` the params here, since the `ArticleSearchForm` only passes on keys that it has defined. But this felt weird so I added an explicit `.permit` even though it reads a bit funny.